### PR TITLE
[PR to mk/tests] CI iteration

### PR DESCRIPTION
**Pull request by nathanhammond**
@mehulkar some of my comments on #1904 are addressed here.

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -17,31 +17,15 @@ jobs:
         os: [ubuntu-latest, macos-latest]
 
     steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 2
-
-      - uses: ./.github/actions/setup-deps
+      - uses: actions/checkout@v
```
Comment (mehulkar) [inline .github/workflows/benchmark.yml:20]:
```suggestion
      - uses: actions/checkout@v3
```

and you want to remove fetch-depth?

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -17,31 +17,15 @@ jobs:
...
-        uses: actions/setup-node@v2
-        with:
-          node-version: 16
-          cache: pnpm
 
       - name: Build
         run: cd cli && make turbo
 
-      - name: Install dependencies
-        run: pnpm install --filter=benchmark
```
Comment (mehulkar) [inline .github/workflows/benchmark.yml:43]: so full `pnpm install`?

File: .github/workflows/pr-go-lint.yml [hunk 3]
```diff
@@ -20,13 +17,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-go
         with:
-          fetch-depth: 2
-
-      - uses: ./.github/actions/setup-deps
-        with:
+          go-version: 1.17
```
Comment (mehulkar) [inline .github/workflows/pr-go-lint.yml:22]: remove this? as mentioned in https://github.com/vercel/turborepo/pull/1904#issuecomment-1244151642

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -33,13 +33,13 @@ jobs:
     steps:
       - uses: actions/checkout@v3
         with:
-          fetch-depth: 2
           token: ${{ secrets.TURBOBOT }}
-
-      - uses: ./.github/actions/setup-deps
+      - uses: ./.github/actions/setup-node
+        with:
+          enable-corepack: false
```
Comment (mehulkar) [inline .github/workflows/release.yml:39]: +1. This was one of the reasons to split up Node and Go for me too!

**Review by mehulkar — APPROVED** [review]
Looks great! one of the reasons I chose _not_ to separate Go/Node is that it started to feel too similar to just `actions/setup-go@v3` , et al. I figured the benefit of sharing *in repo* was that our "abstraction" didn't need to be so clean, and based more on *turborepo* rather than the underlying tools. 

Anywho, this is great and a welcome improvement.

**Comment by mehulkar** [comment]
Merging this in to see how it behaves, thank you!

File: .github/workflows/benchmark.yml [hunk 5]
```diff
@@ -17,7 +17,7 @@ jobs:
         os: [ubuntu-latest, macos-latest]
 
     steps:
-      - uses: actions/checkout@v
+      - uses: actions/checkout@v3
```
Comment (nathanhammond) [inline .github/workflows/benchmark.yml:20]: Good catch! 😅

File: .github/workflows/benchmark.yml (hunk 2 above)
Comment (nathanhammond) [reply .github/workflows/benchmark.yml:43]: Yeah, low cost since it'll come from local cache.

**Pull request merged**